### ISSUE TITLE
ITV Hub issue

### ISSUE THREAD

**Issue by scuttster** [Ascoware/get-iplayer-automator#189]
Have issues this evening try to download programs from itv.
Programs can be found and added to the queue however when starting the download it fails with an error of problem unknown. it was working perfectly fine when last used on the 18th of June but today (20th) no luck

this is the contents of the log:

Get iPlayer Automator 1.13.9 Initialized.

INFO: Loading proxy settings...
INFO: No proxy to load
INFO: Proxy load complete.
INFO: No proxy will be used
GetITVShows: ITV Cache Update Starting 
<AppController: 0x7ff107c2a6e0>: Updating Programme Index Feeds...

GetITVShows (Info): Episodes: 1392 Today Programmes: 185 
<AppController: 0x7ff107c2a6e0>: Index Updated.
GetITVShows: Update Finished
/Applications/Get iPlayer Automator.app/Contents/Resources/get_iplayer
--nocopyright
-e60480000000000000
--type=tv,itv
--listformat=SearchResult|<pid>|<available>|<type>|<name>|<episode>|<channel>|<seriesnum>|<episodenum>|<desc>|<thumbnail>|<web>|<available>
--long
--nopurge
--search
midsomer
--profile-dir=/Users/*****/Library/Application Support/Get iPlayer Automator

INFO: Loading proxy settings...
INFO: No proxy to load
INFO: Proxy load complete.
INFO: No proxy will be used
AppController: Starting Downloads

Downloading Show 1/1:

ITV Download (ID=396005): Downloading Midsomer Murders - Sat Jun 16
INFO: Preparing Request for Auth Info
INFO: Metadata processed.
ITV Download (ID=396005): youtube-dl finished downloading
AppController: Downloads Finished

**Comment by gacres2** [comment]
I have my log on verbose and I get the following as well before the second to last bit in the above log ...
...
ERROR: Unable to download XML: HTTP Error 503: Service Unavailable (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

  File "/Applications/Get iPlayer Automator.app/Contents/Resources/youtube-dl/youtube_dl/extractor/common.py", line 579, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/Applications/Get iPlayer Automator.app/Contents/Resources/youtube-dl/youtube_dl/YoutubeDL.py", line 2211, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 437, in open
    response = meth(req, response)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 550, in http_response
    'http', request, response, code, msg, hdrs)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 475, in error
    return self._call_chain(*args)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 558, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
...

in case that helps at all :)

**Comment by mutantcamel** [comment]
This looks like it's a problem that has been reported a few times before with youtube-dl, the app is failing ITV's browser check, and giving a 503 error.

**Comment by mutantcamel** [comment]
I'm not in the best position to be testing this, but I haven't been able to get this work with the most recent version of youtube-dl.  It looks like on the ITV site, they've ring fenced the whole site and now even require a login for Smart TVs.  I haven't been able to get this to work with either --cookies or adding my username and password to youtube-dl.

**Comment by skovatch** [comment]
It looks like the entire sequence that leads up to getting a streaming playlist has changed. I recall looking at the http traffic to ITV before switching over to youtube-dl, and it's completely different from the last time I looked at it. It does look like something they should be able to sort out, though.

**Comment by nitramlatep** [comment]
Yes I have the same issue. Tried Downie, that works fine.

**Comment by skovatch** [comment]
Okay, looks like youtube-dl has already fixed the problem on their end. I'll pull the change later tonight when I'm not at work and see what happens.

https://github.com/rg3/youtube-dl/commit/30374f4d40d8c993bf92c5af9b9c073da49fe8b9

**Comment by Vangelis66** [comment]
... It looks as though the legacy API that still returned **non geo-blocked** RTMPE streams for some of the ITVHub VOD offerings has been put to sleep... 😡 ; [**[itv] Make SOAP request non fatal**](https://github.com/rg3/youtube-dl/commit/30374f4) took care of that and allows yt-dl to proceed to the new API and the **geo-blocked** (and encrypted) HLS streams it returns...

> youtube-dl has already fixed the problem on their end.
> rg3/youtube-dl@30374f4

https://github.com/rg3/youtube-dl/commit/a4ec451 also comes into play here...

> I'll pull the change later tonight when I'm not at work and see what happens.

I modified my local copy of `itv.py` as per linked commits and with the standalone youtube-dl app 
(Windows 32-bit with Python-3.6.5 32-bit) I tried the programme mentioned by OP: 
`https://www.itv.com/hub/midsomer-murders/396005`

My HTTP UK proxy, for some unexplained reason, aborted the download @0.2%, 
but my SSH Tunnel worked nicely: 
```
youtube-dl --console-title --proxy "socks://localhost:1080" "https://www.itv.com/hub/midsomer-murders/396005" --hls-prefer-native --hls-use-mpegts --no-part --fixup never --write-sub --convert-subs srt -o "ITV_MidsomerMurders_Se8Ep2_DeadInTheWater[396005].ts" => 

[ITV] 396005: Downloading webpage
[ITV] 396005: Downloading XML
WARNING: Unable to download XML: HTTP Error 503: Service Unavailable
[ITV] 396005: Downloading JSON metadata
[ITV] 396005: Downloading m3u8 information
[info] Writing video subtitles to: ITV_MidsomerMurders_Se8Ep2_DeadInTheWater[396
005].en.vtt
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 454
[download] Destination: ITV_MidsomerMurders_Se8Ep2_DeadInTheWater[396005].ts
[download]  47.4% of ~1.01GiB at 377.40KiB/s ETA 22:43
```
Hence, I see no reason why the fix wouldn't work in GiA :wink:
Regards

**Comment by skovatch** [comment]
Fixed in 1.13.10.
